js/src/lib/Details/FileBrowser.js

```diff
@@ -28,8 +24,16 @@ export default class FileBrowser extends React.PureComponent {
 
   _fetchFiles() {
     this.setState({ error: null, files: null });
-    const url = this._getBaseURL() + '?meta';
+    const url = `https://data.jsdelivr.com/v1/package/npm/${this.props.objectID}@${this.props.version}`;
     fetch(url)
+      .then(response => {
+        if (!response.ok) {
```

Comment:
Shouldn't error handling be in ` catch` block, or as the second argument to `then`? Does jsdelivr not use proper HTTP codes for their responses?

Reply:
This was a surprise to me as well. `fetch()` [doesn't throw based on status code](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful) but only if there was a network error. jsDelivr errors have the appropriate status code, and a message in the body. For a unified error handling, we need to parse the body and throw it as an error ourselves.